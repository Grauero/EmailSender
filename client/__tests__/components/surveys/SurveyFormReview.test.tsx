import 'jest';
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import {
  SurveyFormReview,
  mapStateToProps,
  mapDispatchToProps
} from '../../../src/components/surveys/SurveyFormReview';

const props = {
  formValues: [{ name: 'n1', label: 'l1' }],
  onCancel: jest.fn(),
  submitSurvey: jest.fn(),
  history: {} as History
};
const component = shallow(<SurveyFormReview {...props} />);

test('matches snapshot', () => {
  expect(toJSON(component)).toMatchSnapshot();
});

test('clicking Back button calls props.onCancel', () => {
  component
    .find('button')
    .at(0)
    .simulate('click');

  expect(props.onCancel).toHaveBeenCalled();
  expect(props.onCancel).toHaveBeenCalledTimes(1);
});

test('clicking Send Survey button calls props.submitSurvey', () => {
  component
    .find('button')
    .at(1)
    .simulate('click');

  expect(props.submitSurvey).toHaveBeenCalled();
  expect(props.submitSurvey).toHaveBeenCalledTimes(1);
  expect(props.submitSurvey).toHaveBeenCalledWith(
    props.formValues,
    props.history
  );
});

test('maps form value from store to props', () => {
  const props = mapStateToProps({
    auth: false,
    surveys: [],
    form: { surveyForm: { registeredFields: [] } }
  });

  expect(props).toEqual({ formFields: undefined });
});

test('dispatches handleToken from store to props', () => {
  const props = mapDispatchToProps(jest.fn());

  expect(Object.keys(props)[0]).toBe('submitSurvey');
});
