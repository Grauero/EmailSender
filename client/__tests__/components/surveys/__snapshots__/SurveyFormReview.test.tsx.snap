// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="preview"
>
  <h5>
    Confirm your entries
  </h5>
  <div
    key="title"
  >
    <label>
      Survey Title
    </label>
    <div />
  </div>
  <div
    key="subject"
  >
    <label>
      Subject Line
    </label>
    <div />
  </div>
  <div
    key="body"
  >
    <label>
      Email Body
    </label>
    <div />
  </div>
  <div
    key="recipients"
  >
    <label>
      Recipient List
    </label>
    <div />
  </div>
  <button
    className="yellow darken-3 white-text btn-flat form-back"
    onClick={[Function]}
    type="button"
  >
    Back
  </button>
  <button
    className="green btn-flat right white-text form-submit"
    onClick={[Function]}
    type="button"
  >
    Send Survey
    <i
      className="material-icons right"
    >
      email
    </i>
  </button>
</div>
`;
